dsgrid/registry/registry_manager.py

```diff
@@ -221,6 +222,97 @@ def load(
         )
         return cls(params)
 
+    def data_sync(self, project_id, dataset_id, no_prompts=True):
+        """Sync data from the remote dsgrid registry.
+
+        Parameters
+        ----------
+        project_id : str
+            Sync by project_id filter
+        dataset_id : str
+            Sync by dataset_id filter
+        no_prompts :  bool
+            If no_prompts is False, the user will be prompted to continue sync pulling the registry if lock files exist. By default, True.
+        """
+        if not project_id and not dataset_id:
+            raise ValueError("Must provide a dataset_id or project_id for dsgrid data-sync.")
+
+        if project_id:
+            if not self.project_manager.has_id(project_id):
+                raise ValueError(f"No registered project ID = '{project_id}'")
+            project_version = self.project_manager.get_current_version(project_id)
+            config_file = self.project_manager.get_config_file(project_id, project_version)
+            config = ProjectConfig.load(
+                config_file, self.dimension_manager, self.dimension_mapping_manager
+            )
+            if dataset_id:
+                if dataset_id not in config.list_registered_dataset_ids():
+                    raise ValueError(
+                        f"No registered dataset ID = '{dataset_id}' registered to project ID = '{project_id}'"
+                    )
+                datasets = [(dataset_id, str(config.get_dataset(dataset_id).version))]
+            else:
+                datasets = []
+                for dataset in config.list_registered_dataset_ids():
+                    datasets.append((dataset, str(config.get_dataset(dataset).version)))
+
+        if dataset_id and not project_id:
+            if not self.dataset_manager.has_id(dataset_id):
+                raise ValueError(f"No registered dataset ID = '{dataset_id}'")
+            version = self.dataset_manager.get_current_version(dataset_id)
+            datasets = [(dataset_id, version)]
+
+        for dataset, version in datasets:
+            self._data_sync(dataset, version, no_prompts)
+
+    def _data_sync(self, dataset_id, version, no_prompts=True):
+        cloud_interface = self._params.cloud_interface
+        offline_mode = self._params.offline
+        dry_run_mode = self._params.dry_run
+
+        if offline_mode:
+            raise ValueError("dsgrid data-sync only works in online mode.")
+        if dry_run_mode == True:
+            sync = False
+        else:
+            sync = True
+
+        lock_files = list(
+            cloud_interface.get_lock_files(
+                relative_path=f"s3:/{cloud_interface._s3_filesystem._bucket}/configs/datasets/{dataset_id}"
```

Comment:
Do you need to include `s3` here? Isn't that part already encapsulated within the s3 cloud interface? Also, you're referring to `cloud_interface._s3_filesystem._bucket.`, which doesn't seem correct because it has two private variable accesses.